emukit/core/discrete_parameter.py

```diff
@@ -21,13 +21,15 @@ def __init__(self, name: str, domain: Iterable):
         self.name = name
         self.domain = domain
 
-    def check_in_domain(self, x: Union[Iterable, float]) -> bool:
+    def check_in_domain(self, x: Union[np.ndarray, Iterable, float]) -> bool:
         """
         Checks if the points in x are in the set of allowed values
 
         :param x: 1d numpy array of points to check
         :return: A boolean indicating whether each point is in domain
         """
+        if isinstance(x, np.ndarray):
+            x = x.ravel()
```

Comment:
Good points. The original reason why I changed this, was for api unification. Categorical parameters expect a 2-d-array (break for 1-d) here and continuous parameters can handle 1-d and 2-d.
Additionally in most other parts of the library (e.g. user functions) 2-d arrays with shape=(points, features) are used consistently, even for features=1. 

What would be your suggestion here to avoid special cases for the discrete parameter?

Reply:
I extended the methods in this way. Please have a look.